models/activities/notification.go

```diff
@@ -372,16 +372,16 @@ func CountUnread(ctx context.Context, userID int64) int64 {
 // LoadAttributes load Repo Issue User and Comment if not loaded
 func (n *Notification) LoadAttributes(ctx context.Context) (err error) {
 	if err = n.loadRepo(ctx); err != nil {
-		return
+		return err
 	}
 	if err = n.loadIssue(ctx); err != nil {
-		return
+		return err
 	}
 	if err = n.loadUser(ctx); err != nil {
-		return
+		return err
 	}
 	if err = n.loadComment(ctx); err != nil {
-		return
+		return err
 	}
 	return err
```

Comment:
what is the need of `return err` 
we can put  `return nil`

Reply:
well we do report the error back so we need to retur err and not nil here

Reply:
It is also a similar case of https://github.com/go-gitea/gitea/pull/25655#issuecomment-1627379525

> Some code like `return mode, fname, sha, n, err`  and `return sha, typ, size, err` are written as is intentionally, to avoid breaking.

The purpose of it is to avoid changing old code too much (which might break something or cause difficulty for reviewing).

In short: In Golang, `return err` is always right for named return values, but if you want to change it, the developers should always spend more time on thinking about "what value should be returned".